cdap-watchdog/src/main/java/io/cdap/cdap/logging/ErrorLogsClassifier.java

```diff
@@ -64,24 +73,48 @@ public Boolean load(String className) {
           }
         }
       });
+  private final Cache<ErrorClassificationResponseCacheKey,
+      List<ErrorClassificationResponse>> responseCache;
+  private final MetricsCollectionService metricsCollectionService;
 
   private static boolean isFailureDetailsProviderInstance(String className) {
     try {
-      return cache.get(className);
+      return CLASS_CACHE.get(className);
     } catch (Exception e) {
       return false; // Handle any unexpected errors
     }
   }
 
+  /**
+   * Constructor for {@link ErrorLogsClassifier}.
+   */
+  @Inject
+  public ErrorLogsClassifier(MetricsCollectionService metricsCollectionService) {
+    this.metricsCollectionService = metricsCollectionService;
+    responseCache = CacheBuilder.newBuilder().maximumSize(1000)
+        .expireAfterWrite(7, TimeUnit.DAYS).build();
+  }
+
   /**
    * Classifies error logs and returns {@link ErrorClassificationResponse}.
    *
    * @param logIter Logs Iterator that can be closed.
    * @param responder The HttpResponder.
    */
-  public void classify(CloseableIterator<LogEvent> logIter, HttpResponder responder) {
+  public void classify(CloseableIterator<LogEvent> logIter, HttpResponder responder,
+      String namespace, String programId, String appId, String runId) throws Exception {
```

Comment:
Please mark which params are nullable and update javadoc

Reply:
done.